src/tensors/cpu/ruy_adapter.h

```diff
@@ -0,0 +1,534 @@
+/*
+ * This file follows intgemm and is a means of retrofitting ruy into the intgemm based wiring in
+ * `intgemm_interface.h`. ruy is an inference backend used in tensorflow and android deployment and
+ * has an optimized ARM backend for the multiply operations required. Optimized code for quantize,
+ * unquantize, transpose are added separately to connect the multiply library to marian.
+ */
+
+#pragma once
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include "ruy/platform.h"
+#include "ruy/system_aligned_alloc.h"
+
+#if RUY_PLATFORM_NEON
+#include <arm_neon.h>
+#endif
+
+namespace marian {
+namespace cpu {
+namespace integer {
+
+using Index = unsigned int;
+
+/*
```

Comment:
This is already copied elsewhere in Marian source I think?

Reply:
```
$ grep -R "using Index = "
tensors/cpu/wasm_intgemm_interface.h:using Index = uint32_t;
tensors/cpu/ruy_adapter.h:using Index = unsigned int;

$ grep -R "typedef unsigned int Index"
3rd_party/intgemm/intgemm/types.h:typedef unsigned int Index;
```

Ruy is coming in as a replacement for intgemm, so can't include `intgemm.h`.